pyigm/clustering/clustering_field.py

```diff
@@ -0,0 +1,241 @@
+        tbinedges : ndarray
+          Defines the transverse bins in comoving Mpc
+        rbinedges
+          Defines the radial bins in comoving Mpc
+
+        Returns
+        -------
+        Internal updates only, e.g. DgDg, RgRg
+
+        """
+
+        self.tbinedges = tbinedges
+        self.rbinedges = rbinedges
+
+
+        # Gal-gal only
+        if self.galreal is not None:
+            self.XYZ_gal()
+            if self.proper:
+                self.XYZ_proper()
+            self.DgDg = auto_pairs_rt(self.xg, self.yg, self.zg, rbinedges, tbinedges, wrap=self.wrapped)
+            self.RgRg = auto_pairs_rt(self.xgr, self.ygr, self.zgr, rbinedges, tbinedges, wrap=self.wrapped)
+            self.DgRg = cross_pairs_rt(self.xg, self.yg, self.zg, self.xgr, self.ygr, self.zgr, rbinedges, tbinedges,
+                                       wrapped=self.wrapped)
+
+        # Abs-abs only
+        if self.absreal is not None:
+            self.XYZ_abs()
+            if self.proper:
+                self.XYZ_proper()
+            self.DaDa = auto_pairs_rt(self.xa, self.ya, self.za, rbinedges, tbinedges, wrap=self.wrapped)
+            self.RaRa = auto_pairs_rt(self.xar, self.yar, self.zar, rbinedges, tbinedges, wrap=self.wrapped)
+            self.DaRa = cross_pairs_rt(self.xa, self.ya, self.za, self.xar, self.yar, self.zar, rbinedges, tbinedges,
+                                       wrapped=self.wrapped)
+
+
+            '''
+            self.DaDg = cross_pairs_rt(self.xa, self.ya, self.za, self.xg, self.yg, self.zg, rbinedges, tbinedges,
+                                       wrapped=wrapped)
+
+            self.DaRg = cross_pairs_rt(self.xa, self.ya, self.za, self.xgr, self.ygr, self.zgr, rbinedges, tbinedges,
+                                       wrapped=wrapped)
+            self.RaRg = cross_pairs_rt(self.xar, self.yar, self.zar, self.xgr, self.ygr, self.zgr, rbinedges, tbinedges,
+                                       wrapped=wrapped)
+            self.RaDg = cross_pairs_rt(self.xar, self.yar, self.zar, self.xg, self.yg, self.zg, rbinedges, tbinedges,
+                                       wrapped=wrapped)
+            '''
+
+    def XYZ_gal(self):
+        """Calculates X,Y,Z coordinates for the galaxies, real and random
+        from (RA,DEC,Z).
+
+        Internals filled only:
+        xg,yg,zg
+        xgr,ygr,zgr
+        """
+        from pyigm.clustering.coord_utils import radec_to_xyz
+
+        # randomize order
+        #self.galreal = shuffle_rec_array(self.galreal)
+        #self.galrand = shuffle_rec_array(self.galrand)
+
+        # Real galaxies
+        #Rlos = np.array([self.cosmo.comoving_distance(self.galreal.ZGAL[i]) / PC / 1e6 for i in xrange(self.galreal.size)])
+        Rlos = self.cosmo.comoving_distance(self.galreal.ZGAL).to('Mpc').value
+        xyz = radec_to_xyz((self.galreal.RA - self.CRA) * np.cos(self.CDEC * np.pi / 180.),
+                           self.galreal.DEC - self.CDEC).T * Rlos
```

Comment:
Would be worthwhile to adopt a unique comoving coordinate system X, Y, Z that applies to the full sky? In the current implementation it assumes the angles are small... around coordinate (CRA, CDEC).

Reply:
I'll add that as a TODO in radec_to_xyz() and add
doc string warnings about the small angle approx for now.